shell/envvar_cleanup.bash

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env bash
```

Comment:
BTW, I actually think I'm ok with `awk` being a hard dependency for unix systems.  Just as a reference I checked an instance I have running busybox, and awk is part of the busybox binary.  I guess I'd generally consider busybox a good data point for what might be possible to make hard dependencies.